### PR TITLE
Update PublicAPI.Shipped for 17.4

### PR DESCRIPTION
**Pull request by ryzngard**

### CODE REVIEW

File: src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -0,0 +1,100 @@
+Microsoft.CodeAnalysis.Internal.Editing.DeclarationKind
```
Comment (333fred) [inline src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: Have we confirmed with @sharwell that we want to consider this package shipped?

Comment (ryzngard) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: I have not. @sharwell can you comment here if this should be reverted or not?

File: src/CodeStyle/Core/Analyzers/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -1,100 +1 @@
-Microsoft.CodeAnalysis.Internal.Editing.DeclarationKind
```
Comment (jcouv) [inline src/CodeStyle/Core/Analyzers/PublicAPI.Unshipped.txt:1]: This PR is targeting the 17.4 release branch. Don't we usually do this operation in the `main` branch?

Comment (ryzngard) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Unshipped.txt:1]: We will have to do in both branches. We'll port from here back to main

Comment (333fred) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Unshipped.txt:1]: @jcouv no, we usually do this in the release branch and let that flow to main.

File: src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt (hunk 1 above)
Comment (sharwell) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: I'm not entirely sure why these items are declared `public`. They appear to be in an internal namespace as an indication that they are not part of the package's official API. The design of the package is that it ships in stable forms for use, but it's not supported to reference the package APIs in other contexts.

Comment (ryzngard) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: It looks like the `Internal` namespace portion is a lie. The type was marked public and left public for the CODESTYLE layer. Is that not intended?

Comment (ryzngard) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: https://github.com/dotnet/roslyn/blob/release/dev17.4/src/Workspaces/Core/Portable/Editing/DeclarationKind.cs

Comment (ryzngard) [reply src/CodeStyle/Core/Analyzers/PublicAPI.Shipped.txt:1]: I'll just revert these to get this in. We can discuss this as a separate change if needed

**Review by 333fred — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Pull request merged**